spec/lib/submission_builder/ty2024/states/nc/documents/d400_spec.rb

```diff
@@ -33,6 +33,15 @@
         expect(xml.document.at('NCTaxPaid').text).to eq tax_paid.to_s
         expect(xml.document.at('RemainingPayment').text).to eq tax_paid.to_s
       end
+
+      it "correctly fills veteran info for primary" do
+        intake.update(primary_veteran: "yes")
+        expect(xml.document.at('VeteranInfoPrimary').text).to eq "1"
+      end
+
+      it "does not include spouse veteran info" do
+        expect(xml.document.at('VeteranInfoSpouse')).to be_nil
+      end
```

Comment:
[DUST] i think this could go in the previous block since it doesn't require a different setup